src/main/java/com/varchenko/MikeHashMap.java

```diff
@@ -0,0 +1,225 @@
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] buckets;
+    private Node<K, V> current;
+    private int capacity;
+    private int size;
+    private int modCount;
+
+
+    public MikeHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        buckets = new Node[capacity];
+    }
+
+    public MikeHashMap(int capacity) {
+        this();
+        this.capacity = capacity;
+    }
+
+    private void resize() {
+        if (size >= capacity * LOAD_FACTOR) {
+            Node<K, V>[] oldBucket = buckets;
+            buckets = new Node[capacity + (capacity >> 1)];
+            for (Node<K, V> kvNode : oldBucket) {
+                if (kvNode != null) {
+                    put(kvNode.getKey(), kvNode.getValue());
+                }
+            }
+        }
+    }
+
+    private int indexFor(Object k) {
+        if (k != null) {
+            int index = k.hashCode() & capacity - 1;
+            return index != 0 ? index : index + 1;
+        }
+        return 0;
+    }
+
+    private void putNotNull(K key, V value) {
+        Node<K, V> node = buckets[indexFor(key)];
+        if (node.getKey() != null) {
+            if (node.getKey().hashCode() == key.hashCode() && node.getKey().equals(key)) {
+                node.setValue(value);
+            } else {
+                node.setNext(new Node<>(key, value));
+                size++;
+                modCount++;
+            }
+        } else {
+            node.setValue(value);
+        }
+    }
+
+    private Node<K, V> findInNode(Node<K, V> node, Object k) {
+        Node<K, V> currentNode = node;
+        while (currentNode.hasNext()) {
+            if (node.getKey().hashCode() == k.hashCode() && node.getKey().equals(k)) {
+                return node;
+            }
+            currentNode = currentNode.next;
+        }
+        return node;
+    }
+
+    @Override
+    public V get(Object k) {
+        Node<K, V> node = buckets[indexFor(k)];
+        if (node != null) {
+            if (node.getKey() != null) {
+                if (node.getKey().hashCode() == k.hashCode() && node.getKey().equals(k)) {
+                    return node.getValue();
+                }
+            }
+            return findInNode(buckets[indexFor(k)], k).getValue();
+        }
+        return null;
+    }
+
+    @Override
+    public V put(K key, V value) {
+        Node<K, V> node = buckets[indexFor(key)];
+        if (node != null) {
+            putNotNull(key, value);
+            resize();
+        } else {
+            buckets[indexFor(key)] = new Node<>(key, value);
+            resize();
+            size++;
+            modCount++;
+        }
+        return buckets[indexFor(key)].getValue();
+    }
+
+    @Override
+    public V remove(Object key) {
+        Node<K, V> node = buckets[indexFor(key)];
+        if (node.getKey() != null) {
+            if (node.getKey().hashCode() == key.hashCode() && node.getKey().equals(key)) {
```

Comment:
You should check the hash of keys only for Node which we need to remove. But you check the hash of the first Node in the bucket. It's incorrect